orttraining/orttraining/core/framework/ortmodule_graph_builder.cc

```diff
@@ -89,6 +89,10 @@ Status OrtModuleGraphBuilder::Build(const std::vector<std::vector<int64_t>>* inp
 
   ORT_RETURN_IF_ERROR(BuildGradientGraph(x_node_arg_names));
 
+  if (config_.enable_caching) {
+    GetFrontierTensors();
```

Comment:
>GetFrontierTensors [](http://example.com/codeflow?start=4&length=18)

The graph gets transformed after this line of code in InferenceSession:Initialize(). Just want to make sure it is ok to fetch the frontier tensors here.

Reply:
With the current plan for fetching the frontier tensors- as graph outputs, we have to fetch the frontier tensors before we transform the graph at this point. This is okay for the initial implementation that will focus on Cast outputs.

Reply:
Thanks, can you please put this in a comment at line 93?

Reply:
will do if there are more changes else in the next PR